RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h

```diff
@@ -0,0 +1,1005 @@
+#ifdef RFIT_DEBUG
+    Rfit::printIt(cov_sz, "Scatter_cov_line - cov_sz: ");
+#endif
+    Matrix2Nd<N> tmp = Matrix2Nd<N>::Zero();
+    for (u_int k = 0; k < n; ++k) {
+      tmp(k, k) = cov_sz[k](0, 0);
+      tmp(k + n, k + n) = cov_sz[k](1, 1);
+      tmp(k, k + n) = tmp(k + n, k) = cov_sz[k](0, 1);
+    }
+    for (u_int k = 0; k < n; ++k) {
+      for (u_int l = k; l < n; ++l) {
+        for (u_int i = 0; i < std::min(k, l); ++i) {
+          tmp(k + n, l + n) += std::abs(S_values(k) - S_values(i)) * std::abs(S_values(l) - S_values(i)) * sig2_S(i);
+        }
+        tmp(l + n, k + n) = tmp(k + n, l + n);
+      }
+    }
+    // We are interested only in the errors orthogonal to the rotated s-axis
+    // which, in our formalism, are in the lower square matrix.
+#ifdef RFIT_DEBUG
+    Rfit::printIt(&tmp, "Scatter_cov_line - tmp: ");
+#endif
+    ret = tmp.block(n, n, n, n);
+  }
+
+  /*!
+    \brief Compute the covariance matrix (in radial coordinates) of points in
+    the transverse plane due to multiple Coulomb scattering.
+    \param p2D 2D points in the transverse plane.
+    \param fast_fit fast_fit Vector4d result of the previous pre-fit
+    structured in this form:(X0, Y0, R, Tan(Theta))).
+    \param B magnetic field use to compute p
+    \return scatter_cov_rad errors due to multiple scattering.
+    \warning input points must be ordered radially from the detector center
+    (from inner layer to outer ones; points on the same layer must ordered too).
+    \details Only the tangential component is computed (the radial one is
+    negligible).
+ */
+  template <typename M2xN, typename V4, int N>
+  __host__ __device__ inline MatrixNd<N> Scatter_cov_rad(const M2xN& p2D,
+                                                         const V4& fast_fit,
+                                                         VectorNd<N> const& rad,
+                                                         double B) {
+    constexpr u_int n = N;
+    double p_t = std::min(20., fast_fit(2) * B);  // limit pt to avoid too small error!!!
+    double p_2 = p_t * p_t * (1. + 1. / (fast_fit(3) * fast_fit(3)));
+    double theta = atan(fast_fit(3));
+    theta = theta < 0. ? theta + M_PI : theta;
+    VectorNd<N> s_values;
+    VectorNd<N> rad_lengths;
+    const Vector2d o(fast_fit(0), fast_fit(1));
+
+    // associated Jacobian, used in weights and errors computation
+    for (u_int i = 0; i < n; ++i) {  // x
+      Vector2d p = p2D.block(0, i, 2, 1) - o;
+      const double cross = cross2D(-o, p);
+      const double dot = (-o).dot(p);
+      const double atan2_ = atan2(cross, dot);
+      s_values(i) = std::abs(atan2_ * fast_fit(2));
+    }
+    computeRadLenUniformMaterial(s_values * sqrt(1. + 1. / (fast_fit(3) * fast_fit(3))), rad_lengths);
+    MatrixNd<N> scatter_cov_rad = MatrixNd<N>::Zero();
+    VectorNd<N> sig2 = (1. + 0.038 * rad_lengths.array().log()).abs2() * rad_lengths.array();
+    sig2 *= 0.000225 / (p_2 * sqr(sin(theta)));
```

Comment:
sorry I'm probably silly, but can you remind me the source of this `0.000225` ? thanks